host_core/native/hostcore_wasmcloud_native/src/oci.rs

```diff
@@ -112,3 +140,17 @@ async fn pull(
         .await
         .map_err(|e| format!("{}", e).into())
 }
+
+async fn cache_oci_image(
+    imgdata: &oci_distribution::client::ImageData,
+    cache_filename: &PathBuf,
+) {
+    let mut f = tokio::fs::File::create(&cache_filename).await.unwrap();
```

Comment:
These unwraps aren't safe (e.g. what if the directory doesn't exist or the host doesn't have permissions to create a file in it). You'll need to handle the error cases. Since the operations here can fail, I think the function should return a `std::io::Result<()>`. That would let you use `?` instead of `unwrap` on these, since it looks like they're all tokio fs calls which return `std::io::Result` already